test_pull_requests

```diff
@@ -1587,23 +1587,55 @@ popd
       URI(build_url + "/api/json")
     end
 
-    def self.is_previous_build_running?(branch, settings)
-      build_num = get_next_build(branch, settings) - 1
-      build_url = Properties['jenkins_url'] + 'job/' + branch_settings(branch, settings)['jenkins_job_name'] + "/#{build_num}/"
+    def self.previous_build_running?(branch, settings, repo)
+      addtl_jenkins_params = branch_settings(branch, settings)['addtl_jenkins_params']
+      previous_running = false
+      consecutive_not_running_count = 0
+      build_num = get_next_build(branch, settings)
       get_session(branch, settings).start do |http|
-        request = Net::HTTP::Get.new to_api_url(build_url).request_uri
-        response = http.request request
-        if response.kind_of? Net::HTTPSuccess
-          return JSON.parse(response.body)["building"]
-        elsif response.kind_of? Net::HTTPNotFound
-          return false # maybe someone deleted the last build?
-        else
-          return true # assume it's running to avoid errors causing multiple builds to start
+        # Search for a running build of the same variety.  Don't run forever.
+        100.times do
+          # Break if you found one or if you run across a bunch of consecutive builds that aren't running you can assume you reached the end
+          break if previous_running || consecutive_not_running_count >= 10
+          build_num -= 1
```

Comment:
OK, this seems to get us what we want, we just have to filter on `building == true` in the script. If we wanted to consume XML, we could filter that on the server side.
(replace `test_pull_requests_origin_networking` with what job you're looking for, obviously)
https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_networking/api/json?tree=builds[building,actions[parameters[name,value]]]